### PR TITLE
[TS] Allow configuration of lower batch size

### PR DESCRIPTION
**Pull request by gmaretic**
**Goals (and why)**:
With autotuning we attempt to construct batches of up to 100k entries. This is generally what we want, as if there are so many entries we want to catch up as soon as possible. However it can take ~30min to process such an iteration, while sometimes we prefer to be slightly slower but able to be more uniformly up to date.

**Implementation Description (bullets)**:
Mostly wiring

**Testing (What was existing testing like?  What have you done to improve it?)**:
Some tests to verify it works as expected, and that sweep delay treats it sanely

**Concerns (what feedback would you like?)**:
Did I miss something?

**Where should we start reviewing?**:
Small

**Priority (whenever / two weeks / yesterday)**:
asap

### CODE REVIEW

**Review by jeremyk-91 — COMMENTED** [review]
I'm broadly happy with the implementation, though let's get a bit of time tomorrow to discuss some of the tradeoffs/implications.

**Comment by gmaretic** [comment]
On the fence if adding this config is that useful as opposed to just modifying the batch size to a fixed 10k. Given how long it takes to run one iteration of 100k entries, seems the overhead is low enough to do it.

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueReader.java [hunk 1]
```diff
@@ -50,4 +50,16 @@ SweepBatchWithPartitionInfo getNextBatchToSweep(ShardAndStrategy shardStrategy,
         }
         return accumulator.toSweepBatch();
     }
+
+    @Immutable
+    public interface ReadBatchingRuntimeConfig {
```
Comment (jeremyk-91) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueReader.java:55]: minor nits: config objects typically have just values. I'd suggest we either pass `Supplier<RBRC>` and use some `Suppliers.transform(...)` higher up, or rename this `ReadBatchingRuntimeContext` or something like that

**Review by jeremyk-91 — APPROVED** [review]
Looks broadly correct, and we spoke externally so I have more context on why this PR is useful. One comment on naming which I found to be a little counterintuitive

**Comment by gmaretic** [comment]
👍

**Pull request merged**